### PR TITLE
Permission Groups

### PR DESCRIPTION
**Pull request by mjoyce91**
Fetches the user's permission groups along with their profile and combines the two objects into one. Then uses this data to determine whether to display the Glossary Editor link in the profile menu.

### CODE REVIEW

File: src/Components/ProfilePage/ProfilePage.jsx [hunk 1]
```diff
@@ -9,10 +9,11 @@ import BidStatistics from '../../Containers/BidStatistics';
 import GlossaryEditor from '../../Containers/GlossaryEditor';
 import ProfileMenu from '../ProfileMenu';
 import { USER_PROFILE } from '../../Constants/PropTypes';
+import { userHasPermissions } from '../../utilities';
 
 const ProfilePage = ({ user }) => (
   <div className="profile-page">
-    <ProfileMenu isCDO={user.is_cdo} />
+    <ProfileMenu isCDO={user.is_cdo} isGlossaryEditor={userHasPermissions(['glossary_editors'], user.permission_groups)} />
```
Comment (burgwyn) [inline src/Components/ProfilePage/ProfilePage.jsx]: Let's list our permissions in `Constants`

File: src/utilities.js [hunk 2]
```diff
@@ -314,3 +314,14 @@ export const formatIdSpacing = (id) => {
   // if id is not defined, return null
   return null;
 };
+
+// provide an array of permissions to check if they all exist in an array of user permissions
+export const userHasPermissions = (permissionsToCheck = [], userPermissions = []) => {
+  let hasPermissions = true;
```
Comment (burgwyn) [inline src/utilities.js]: Let's invert this.  User should be assumed to NOT have a permission.

Comment (mjoyce91) [reply src/utilities.js]: I found a more straight forward way to check:

```js
permissionsToCheck.every(val => userPermissions.indexOf(val) >= 0)
```

File: src/utilities.test.js [hunk 3]
```diff
@@ -395,3 +396,42 @@ describe('formatIdSpacing', () => {
...
+    expect(userHasPermissions(permissionsToCheck, userPermissions)).toBe(false);
+  });
+
+  it('returns true if the user has more than the needed permissions', () => {
+    userPermissions.push('d', 'e');
+    expect(userHasPermissions(permissionsToCheck, userPermissions)).toBe(true);
+  });
+
+  it('returns false if userPermissions are not provided', () => {
+    userPermissions = [];
```
Comment (burgwyn) [inline src/utilities.test.js]: I almost feel like we should throw an exception in this case as it might be a sign that the user's session is stale

**Pull request merged**